src/poliastro/plotting/orbit/backends/matplotlib_backend.py

```diff
@@ -0,0 +1,509 @@
+"""A module implementing matplotlib backend for orbit plotters."""
+
+from poliastro.plotting.orbit_plotters.backends.base import (
+    Trajectory,
+    _OrbitPlotterBackend,
+)
+
+
+class OrbitPlotterMatplotlib2D(_OrbitPlotterBackend):
```

Comment:
I was thinking that we could have something like this:

```python
class OrbitPlotterMatplotlib2D:  # Notice it doesn't inherit from the backend
    def __init__(self):
        self._backend = _MatplotlibBackend2D(...)

    def plot_coordinates(self, ...):
        # Prepare data
        ...
        self._backend.draw_line(...)  # What now is called "draw_coordinates"
        # Wrap up
        ...
```

in sum:

- A plotter _has_ a backend (rather than "a plotter _is_ a backend") - i.e. [composition over inheritance](http://en.wikipedia.org/wiki/Composition_over_inheritance)
- The plotter takes care of preparing the inputs and the backend only takes care of drawing primitives (`draw_line` for coordinates, `draw_sphere` for bodies, `draw_arrow` for impulses)

I think this should decouple the logic in a more effective way and avoid having too many methods.

Reply:
Really like the approach you presented in here. I was over complicating things.